sqa/starknet/writer/writer.py

```diff
@@ -104,17 +122,114 @@ def write_parquet(loc: Fs, tables: dict[str, pyarrow.Table]) -> None:
             'from_address',
             'block_number',
             'transaction_index',
-            'event_index'
+            'event_index',
         ],
-        **kwargs
+        **kwargs,
     )
 
     LOG.debug('wrote %s', loc.abs('events.parquet'))
 
+    # Handling Starknet traces
+    traces = tables['traces']
+    traces = traces.sort_by([
+        ('block_number', 'ascending'),
+        ('transaction_index', 'ascending'),
+    ])
+
+    traces = add_index_column(traces)
+
+    loc.write_parquet(
+        'traces.parquet',
+        traces,
+        use_dictionary=[
+            'trace_type',
+            'call_type',
+            'invocation_type',
+        ],
+        row_group_size=50_000,
+        write_statistics=[
+            '_idx',
+            'block_number',
+            'transaction_index',
+        ],
+        **kwargs,
+    )
+
+    LOG.debug('wrote %s', loc.abs('traces.parquet'))
+
+    # Handling Starknet call messages
+    messages = tables['messages']
+    messages = messages.sort_by([
+        ('block_number', 'ascending'),
+        ('transaction_index', 'ascending'),
```

Comment:
transaction_index is not a part of primary key for messages. so should be excluded

Reply:
transaction_index can't be excluded from messages primary key
block_number -> transaction_index -> order / trace_address

Reply:
but message order is calculated per block so we can safely omit tx_index

Reply:
I was not able to confirm, still removed transaction_index from primary key